### PR TITLE
MAISTRA-1717 Use go-based yq in generate-manifests.sh

### PR DESCRIPTION
**Pull request by luksa**

### CODE REVIEW

**Comment by dgn** [comment]
I think [Jonh made some changes that resulted in the `yq` binaries being called `yq-go` and `yq-python`](https://github.com/maistra/test-infra/pull/114) (which is why the gen-check job is failing here). If this removes the need for the python variant, we should remove that logic from the Dockerfile

**Comment by luksa** [comment]
The go-based yq tool always indents list entries (there's no config option to stop it from doing this), so the resulting YAMLs have slightly different whitespace. Also, some empty strings are encoded as `""` instead of `''`. The same goes for boolean values (e.g. `"true"` instead of `'true'`). 

The nice benefit of using the go-based yq tool is that it preserves comments in the YAML.

**Comment by luksa** [comment]
@dgn Thanks for the tip. I changed the script to first look for `yq-go`, which should make gen-check work now.

File: build/generate-manifests.sh [hunk 1]
```diff
@@ -40,22 +40,18 @@ fi
...
-  # Note there's another yq package, written in golang, which is not the one we want: https://github.com/mikefarah/yq
-  YQ=$(which yq-python 2>/dev/null || which yq 2>/dev/null || echo "")
-
+  # Looks for go-based yq (https://github.com/mikefarah/yq) first, as installed in CI image.
+  # Note there's another yq package, written in python, which is not the one we want: https://github.com/kislyuk/yq
+  YQ=$(which yq-go 2>/dev/null || which yq 2>/dev/null || echo "")
+  echo "Using yq: ${YQ}"
   if ! [ -x "$(command -v ${YQ})" ]; then
-    echo "Please install the python yq package, e.g. 'pip install --user yq'"
+    echo "Please install the go-based yq package (https://github.com/mikefarah/yq/releases/tag/v4.6.0)"
```
Comment (rcernich) [inline build/generate-manifests.sh]: Can we just add this as a tool and invoke it using `go run`?

Comment (luksa) [reply build/generate-manifests.sh]: Best idea ever! Done!

**Comment by dgn** [comment]
I'm not sure how I feel about adding all these things into `vendor/` as they're not runtime dependencies. Is there a way we can `go run` it without that?

**Comment by luksa** [comment]
We've already done it like this for the other build-time dependencies. We could probably not vendor these things, but don't we need to ensure that the build is 100% repeatable and thus vendoring the build-time stuff also makes sense.

**Comment by jwendell** [comment]
+1 on Daniel's comment. What's the problem with relying on yq being present on the host (like the previous iteration of this PR)?

**Comment by luksa** [comment]
@rcernich suggested we use `go run` as we already do for `deepcopy-gen`, `controller-gen`, `conversion-gen`. This way you just run make and everything works.

File: build/generate-manifests.sh [hunk 2]
```diff
@@ -40,25 +40,7 @@ fi
...
-    echo "Please install the python yq package, e.g. 'pip install --user yq'"
-    exit 1
-  else
-    s="yq 2.*"
-    if ! [[ $(${YQ} --version) =~ $s ]]; then
-      echo "Install the correct (python) yq package, e.g. 'pip install --user yq'"
-      exit 1
-    fi
-  fi
+  :
```
Comment (jwendell) [inline build/generate-manifests.sh:43]: You can remove this function

**Review by jwendell — COMMENTED** [review]
My concern is that we end up like envoy, vendoring everything but the basic compiler.

File: build/generate-manifests.sh (hunk 2 above)
Comment (luksa) [reply build/generate-manifests.sh:43]: I left it in in case we need to add dependencies later. But I'll remove it.

**Comment by brian-avery** [comment]
Yeah, I'm of two minds here. On the one hand, it installs everything it requires. On the other hand, it's adding extra stuff. We handle the dependencies via docker, but if you don't have docker installed this may cause issues. Is there any risk with the go mod solution if you already have, say, the python version installed?

**Comment by dgn** [comment]
> Yeah, I'm of two minds here. On the one hand, it installs everything it requires. On the other hand, it's adding extra stuff. We handle the dependencies via docker, but if you don't have docker installed this may cause issues. Is there any risk with the go mod solution if you already have, say, the python version installed?

No, and that's really the big advantage of doing it like this.

I think I'm fine with merging this as-is, but we should then use the same solution in maistra.github.io

**Review by dgn — APPROVED** [review]

**Comment by brian-avery** [comment]
Nevermind. Go run will take care of my concern.

**Review by brian-avery — APPROVED** [review]

**Pull request merged**